### PR TITLE
🎉 Show folders in views! 🎉

### PR DESCRIPTION
**Pull request by cal2195**
This _cool_ PR adds the option to show folders in the `thumbnails`, `files` and `clips` view! 🎉 

It basically replicates a standard file browser, which a lot of people will probably like - and the ability to browse to a folder, then flip the folder view off and continue like normal is _very_ cool! 😄 

It has some pretty snazzy features, which I'll leave you to try it to see what you think! 😉 

I also removed the `folder only` view, as I don't think it added anything with this option now - I can revert that commit if you think otherwise! 👍 

**Caveats**:
- `Folder Intersection` is treated as an exact prefix match if the first entry starts with `/`

### CODE REVIEW

**Comment by cal2195** [comment]
Update: Works on Windows here for me, but only because I created the library on Linux - see #131

**Comment by cal2195** [comment]
I haven't tested this on Windows yet, but this should now work with either an existing library after doing a meta regen in the settings, or on just a new library with this version! 👍 

I've also added support for sorting with folders enabled! 😄 🎉 Along with folder contents details!

**Comment by cal2195** [comment]
Ready for review! 👍

**Comment by cal2195** [comment]
These are going to have a few merge conflicts, so whenever you get a chance to check them out just let me know which you're good with and I'll address the conflicts and merge them! 👍

File: src/app/components/home/thumbnail/preview.component.ts [hunk 1]
```diff
@@ -51,8 +54,30 @@ export class PreviewComponent implements OnInit {
...
+      this.numberOfThumbs = this.folderThumbPaths.length;
+    } else {
+      this.firstFilePath = 'file://' + this.folderPath + '/' + 'vha-' + this.hubName + '/thumbnails/' + this.video.hash + '.jpg';
+      this.fullFilePath =  'file://' + this.folderPath + '/' + 'vha-' + this.hubName + '/filmstrips/' + this.video.hash + '.jpg';
+      this.folderThumbPaths.push(this.firstFilePath);
+      this.numberOfThumbs = 1;
+    }
+  }
+
+  shuffle(a) {
```
Comment (whyboris) [inline src/app/components/home/thumbnail/preview.component.ts]: This method appears exactly the same in `clip.component` which is fine for now, but a few minor recommendations:
- include `JSDoc` to briefly describe what the method does
- add TypeScript to the input (so it's clear what kind of objects are allowed to be used as input)
- place the code for this method into some helper `service` and refer to the method from both places

All three can be accomplished later after the merge 👌

**Comment by whyboris** [comment]
❤️ OMG you're amazing 🌟 

This is _so_ cool! 🙇

**Pull request merged**